metadata-ingestion/src/datahub/utilities/sqllineage_patch.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+from sqllineage.core.holders import SubQueryLineageHolder
+from sqllineage.core.models import Column
+from sqllineage.exceptions import SQLLineageException
+from sqllineage.utils.constant import EdgeType
+
+
+# Patch based on sqllineage v1.3.3
+def end_of_query_cleanup_patch(self, holder: SubQueryLineageHolder) -> None:  # type: ignore
+    for i, tbl in enumerate(self.tables):
+        holder.add_read(tbl)
+    self.union_barriers.append((len(self.columns), len(self.tables)))
+
+    for i, (col_barrier, tbl_barrier) in enumerate(self.union_barriers):
+        prev_col_barrier, prev_tbl_barrier = (
+            (0, 0) if i == 0 else self.union_barriers[i - 1]
+        )
+        col_grp = self.columns[prev_col_barrier:col_barrier]
+        tbl_grp = self.tables[prev_tbl_barrier:tbl_barrier]
+        tgt_tbl = None
+        if holder.write:
+            if len(holder.write) > 1:
+                raise SQLLineageException
+            tgt_tbl = list(holder.write)[0]
+        elif holder.read:
+            tgt_tbl = list(holder.read)[0]
```

Comment:
For all `i`, is the `tgt_tbl` always expected to be the same? If now, how are the header.read/write getting updated, since we always seem to take the value at index 0.